### PR TITLE
[JENKINS-72577] Revert "Increase MavenTest#sensitiveParameters() timeout to 500s"

### LINKED ISSUE #3890: Possible change to ci.jenkins.io's linux agents causes core test failure
### Service(s)

ci.jenkins.io

### Summary

Hey,

It appears that a change to the linux agents with Java 17 and 21 caused a test failure in Jenkins core. Around January the 5fth, `MavenTest#sensitiveParameters()` started to take much longer than usual.

Starting with [5730](https://ci.jenkins.io/job/Core/job/jenkins/job/master/5730/consoleText) the single test takes 229s (!!), exceeding our test timeout of 180s by far, causing builds to fail.

[5728](https://ci.jenkins.io/job/Core/job/jenkins/job/master/5728/consoleText) notes ~39s for the entire `MavenTest`, build on January the 3rd.

Locally, the entire `MavenTest` takes less than 30s to complete for me using `mvn clean verify -Dtest=hudson.tasks.MavenTest`.

Interestingly, this affects linux agents with Java 17 and 21 only. The test passes in its usual time on windows agents with Java 17 and linux agents with Java 11: https://ci.jenkins.io/job/Core/job/jenkins/job/master/5730/pipeline-graph/

### Reproduction steps

_No response_

### PR DESCRIPTION
**Pull request by dduportal**
See [JENKINS-72577](https://issues.jenkins.io/browse/JENKINS-72577).

Reverts jenkinsci/jenkins#8840 which was a workaround for slow tests in the ci.jenkins.io's agents infrastructure as described in https://github.com/jenkins-infra/helpdesk/issues/3890.

In the past 4 months, a lot of changes had been applied to the underlying infrastructure such as:

- JDK21 upgrade to 21.0.3+9
- Underlying Linux machines upgraded to powerful ones
- Underlying network fixed
- Agents workload fully migrated to an unified Azure cloud
- etc.

Fixes https://github.com/jenkins-infra/helpdesk/issues/3890

### Testing done

I can see that the last "main" branch test (see https://ci.jenkins.io/job/Core/job/jenkins/job/master/6227/testReport/hudson.tasks/MavenTest/) is around ~6s on Linux (on ALL JDKs) and ~13s on Windows (JDK17):

Capture d’écran 2024-06-11 à 10 55 43

### Proposed changelog entries

- Remove test timeout workaround added in jenkinsci/jenkins#8840 

### Proposed upgrade guidelines

N/A

### Desired reviewers

@Not-NotMyFault @basil 

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by basil — APPROVED** [review]
Many thanks for the performance improvements on the underlying infrastructure that have allowed this temporary workaround to be reverted.

**Review by timja — APPROVED** [review]
Thanks I removed the timeout which was added for the same issue it was just increased once

----

/label ready-for-merge

----

This PR is now ready for merge, after the build has passed, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
